modules/packages/RangeChunk.chpl

```diff
@@ -84,6 +84,43 @@ module RangeChunk {
       else start..end;
   }
 
+  /*
+    Iterates through a range ``r``, which is blocked up in repeating ```numInds```
+    blocks of size ```blockSize```. Blocks are indexed from 0..numInds-1. 
+    Emits block at index ```idx``` in a cyclic manner.
+  */
+  iter blockCyclicChunks(r: range(?t, boundedType=BoundedRangeType.bounded, ?strided), blockSize: integral, idx: integral, numInds: integral){
+    if (idx >= numInds) then
+      halt("Parameter idx must be < numInds because blocks are indexed from 0..numInds-1");
+
+    if (blockSize <= 0) then
+      halt("blockSize must a positive number");
+
+    if (numInds <= 0) then
+      halt("numInds must be a positive number");
+
+    var range_stride = r.stride;
```

Comment:
This doesn't seem to work quite right for some negative strided ranges. For example the first range yielded by
`blockCyclicChunks(111..1010 by -2, 3, 0, 3)`
is:
`1005..1010 by 2`

but 1005 isn't in the original range because negative stride means start counting from the high index.  It might be simplest to not allow negative strides.

Reply:
@daviditen it should be ```1010..1005 by -2```, but I wanted to keep the output consistent, so I only used positive strides, hence ```1005..1010 by 2```

Reply:
But I can it the other way around if you want, it's only a matter of flipping the output ```low..high by abs(stride)``` to ```high..low by stride```

Reply:
`1010..1005 by -2` is an empty range because the low bound is bigger than the high bound. `1005..1010 by -2` is equivalent to `1006..1010 by -2` - either one represents the values 1006, 1008, 1010. On the other hand, `1005..1010 by 2` represents the values 1005, 1007, 1009.

Reply:
Oh, you're right. I didn't see that. So, ```low..high by stride```? Which would make it to be ```1005..1010 by -2```?

Reply:
Yes, `1005..1010 by -2` or `1006..1010 by -2` is what I'd expect.